### PR TITLE
Sortable no longer depends on Underscore

### PR DESCRIPTION
**Pull request by sloria**
- Add simplified implementation of filter and difference
- Use Array.prototype.map instead of _.map
- Use Array.prototype.forEach instead of _.each
- Update docs

### CODE REVIEW

File: js/sortable.jsx [hunk 1]
```diff
@@ -24,6 +22,25 @@ var userSelect = rhs => {
...
+        if (predicate(value, i, obj))  results.push(value);
+    }
+    return results;
+}
+
+/**
+ * Return the values from arr1 that are not present in arr2.
+ */
+function difference(arr1, arr2) {
+    return filter(arr1, (each) => {return arr2.indexOf(each) < 0});
```
Comment (kevinb7) [inline js/sortable.jsx]: Can you rename "each" to "item"?

**Comment by kevinb7** [comment]
@sloria I think in general it makes sense to remove dependencies so that react-components is more lightweight.  I haven't actually used this library, is it possible to require this component on its own?

**Comment by spicyj** [comment]
Another thing we could consider is depending on `lodash.difference` which only pulls in that one function. +1 for builtin map and forEach though.

File: js/sortable.jsx (hunk 1 above)
Comment (sloria) [inline js/sortable.jsx]: Done. Let me know if you'd like me to squash the commit for that change.

**Comment by sloria** [comment]
@spicyj `lodash.difference` might still be more LOC than it's worth. It includes code to handle different input types and variable arity--the sortable component has just two usages of `difference` for which both the input types and arity are always the same.

Another downside is that if you are already using the full `lodash` package in your project, depending on `lodash.difference` will add duplicated code to your vendor bundle.

**Comment by spicyj** [comment]
Your approach adds duplicated code to your bundle regardless of whether you're using `lodash`. There's not a practical difference between 10 and 50 lines here and I'm much happier to depend on a well-maintained library like lodash rather than reimplementing things and having the potential for bugs.

File: js/sortable.jsx [hunk 2]
```diff
@@ -24,6 +22,25 @@ var userSelect = rhs => {
 
 var sortableDisabled = userSelect("none");
 
+/**
+ * Behaves like Array.prototype.filter, except it handle any iterable type (not just Arrays).
```
Comment (spicyj) [inline js/sortable.jsx]: In any case, this comment is not true. It works on any array-like object (something with a `.length`, etc), not any iterable (like generators, Sets, and other objects with `Symbol.iterator`).

And for array-like objects. you can already use `Array.prototype.filter.call(nodes, predicate)` and don't need to reimplement.

**Comment by spicyj** [comment]
If we use Array.prototype.filter I guess the one-line `difference` function here is fine.

**Comment by sloria** [comment]
@spicyj I originally used `Array.prototype.filter`, but this breaks because we need to filter a `NodeList` [here](https://github.com/sloria/react-components/blob/440725160c1603bd7a23f3ef329f572b5d7a2019/js/sortable.jsx#L129), which doesn't implement `filter`.

**Comment by spicyj** [comment]
Did you see my other comment? `Array.prototype.filter.call(nodeList, pred)` should work fine.

**Comment by sloria** [comment]
@spicyj Ah, sorry, yeah, just caught your other comment. Yes, that will work; I'll change the code.

**Comment by kevinbarabash** [comment]
@sloria these changes have been made by another commit.  Sorry for not merging this sooner.

**Comment by sloria** [comment]
@kevinbarabash No problem. Glad to see the change go in.

**Pull request closed without merging**